interview-questions.md

```diff
@@ -8,36 +8,167 @@ Then, research the question to expand on your answer. Even if you feel you have
+(source -https://guides.rubyonrails.org/active_record_migrations.html)
 
 2. Which RESTful routes must always be passed params? Why?
 
-Your answer:
+Your answer: show, update, edit, destroy all require params passed with them because all of these routes call on a specific instance of data within a model to manipulate. params allow us to access specific columns of a model. we dont need params for new, index, or create because they dont need a specific piece of data to perform their methods.
 
-Researched answer:
+Researched answer: We pass params into routes to essentialy require extra information from a user in order to access specific instances of data within a model. We can also use params when creating new instances of data within a model, but it isnt required in the routing specifically. All of the show, edit, update, and destroy routes need params in the routing in order to perform their respective controller methods.
+(source - https://guides.rubyonrails.org/routing.html, https://www.rubyguides.com/2019/06/rails-params/)
 
 3. Name three rails generator commands. What is created by each?
 
 Your answer:
 
-Researched answer:
+rails g model Model_name column_name:data_type column_name:data_type - generates a model that has a table with as many columns as you define. - creates a model.rb and controller.rb (nope) for the model - also creates a migration file and updates/adds the schema file (i'm not actually sure if rails creates a schema at first???) - creates a spec file if running rspec
+
+rails g resource Model_name column_name:data_type column_name:data_type - does all of the above, but also creates the generic RESTful routes in the routes.rb file
+
+rails g migration whatever_you_want_to_change - creates a migration file that is titled whatever you named the migration - its a file that you can then define a method that changes or adds things to an existing model
+
+Researched answer: Three common rails generator commands are Model, Resource, and Controller. The model generator creates a migration, model and tests for the files that where generated. A resource generator does the same as the model generator, but also creates the controller, resource routes, and a views folder. the Controller generator creates a (you guessed it) controller as well as methods if you chose to create specific methods when running the controller generator. We can also generate migrations, which create migration files. There is another generator i've found called Scaffold that creates a model, controller, and all the views needed for restful routing and full crud functionality (why even code at this point lol). Its not considered best practice to use Scaffold because it takes too much power away from the developer when creating a model.
```

Comment:
Definitely want to keep in mind which files we are creating and if they are necessary. Nice job👏🏽